src/Todoist.Net/Models/Duration.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+
+using Newtonsoft.Json;
+
+namespace Todoist.Net.Models
+{
+    /// <summary>
+    /// Represents durations for tasks.
+    /// </summary>
+    public class Duration
+    {
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Duration" /> class.
+        /// </summary>
+        /// <param name="amount">The time amount.</param>
+        /// <param name="unit">The time unit.</param>
+        public Duration(int amount, DurationUnit unit)
+        {
+            Amount = amount > 0
+                ? amount
+                : throw new ArgumentOutOfRangeException(nameof(amount), "Parameter must be greater than zero.");
+            Unit = unit
+                ?? throw new ArgumentNullException(nameof(unit));
+        }
+
+        internal Duration()
+        {
+        }
+
+        /// <summary>
+        /// Gets or sets the duration time amount.
+        /// </summary>
+        /// <remarks>
+        /// Must be a positive integer (greater than zero).
+        /// </remarks>
+        /// <value>
+        /// The time amount.
+        /// </value>
+        [JsonProperty("amount")]
+        public int Amount { get; set; }
+
+        /// <summary>
+        /// Gets or sets the duration time unit.
+        /// </summary>
+        /// <remarks>
+        /// Either <c>minute</c> or <c>day</c>.
+        /// </remarks>
+        /// <value>
+        /// The duration unit.
+        /// </value>
+        [JsonProperty("unit")]
+        public DurationUnit Unit { get; set; }
+
+
+        /// <summary>
+        /// Gets the value of the duration as a <see cref="TimeSpan"/> object.
+        /// </summary>
+        /// <value>
+        /// The <see cref="TimeSpan"/> value of the duration.
+        /// </value>
+        [JsonIgnore]
+        public TimeSpan TimeValue
+        {
+            get
+            {
+                if (Unit == DurationUnit.Minute)
+                {
+                    return TimeSpan.FromMinutes(Amount);
+                }
+                if (Unit == DurationUnit.Day)
+                {
+                    return TimeSpan.FromDays(Amount);
+                }
+                throw new NotImplementedException();
```

Comment:
I'd suggest to change ifs to switch pattern in this way static analyzers will find an issue if we add a new DurationUnit in the future.

Reply:
You have a good point, but the problem is that we are dealing with static values that won't work with switch patterns.
I've looked it up and found a way to make the code more readable, following this [SO answer](https://stackoverflow.com/a/49624848/6668824), I did the following:
```cs
switch (Unit)
{
    case var _ when Unit == DurationUnit.Minute:
        return TimeSpan.FromMinutes(Amount);
    case var _ when Unit == DurationUnit.Day:
        return TimeSpan.FromDays(Amount);
    default:
        throw new NotImplementedException();
}
```
Unfortunately, this still doesn't guide static analyzers to help us add support for new values.

I thought of adding an `enum` that can be mapped to each of the `DurationUnit` values, but again, analyzers will be useful only if new values are added both to the class and the enum.

Maybe there's something that I'm missing, what do you think?